src/iso_alloc.c

```diff
@@ -349,7 +349,14 @@ INTERNAL_HIDDEN void _iso_alloc_destroy_zone_unlocked(iso_alloc_zone_t *zone, bo
          * back to the OS. It will still be available if we
          * try to use it */
         madvise(zone->bitmap_start, zone->bitmap_size, MADV_DONTNEED);
+#if !__APPLE__
         madvise(zone->user_pages_start, ZONE_USER_SIZE, MADV_DONTNEED);
+#else
+        /* we allow the system to better track user_pages_start here
+         * see MADV_FREE_REUSE note */
+        while(madvise(zone->user_pages_start, ZONE_USER_SIZE, MADV_FREE_REUSABLE) == -1 && errno == EAGAIN) {
```

Comment:
Why the while loop? The calls to `madvise` above can fail and we just continue as their success is not required to keep functioning. I'd prefer to set the flags based on OS ifdef and then have a single call to `madvise`.

Reply:
Note the while loop occurs only in case of EAGAIN.
I found another example [here](https://github.com/adobe/webkit/blob/master/Source/WTF/wtf/OSAllocatorPosix.cpp#L51)

Reply:
Ah I see now. Thanks for pointing that out.